docs/troubleshooting.md

```diff
@@ -0,0 +1,74 @@
+
+        * Decoding the cloud env var for the users: https://docs.microsoft.com/en-us/azure/storage/common/storage-powershell-independent-clouds#get-endpoint-using-get-azenvironment 
+
+    * create `azureoperatorsettings.yaml` file from step-6 mentioend here: https://operatorhub.io/operator/azure-service-operator replace the placeholder values into the respective field and do `kubectl apply -f <location of your azureoperatorsettings.yaml fil>`
+
+    * User should be able to see `azureoperator-controller-manager-***` pods running in there cluster. helpful command to see the pod: `kubectl get pods -n operators` 
+
+## I have already installed `azureserviceoperator` but unsure what service principal ceret is in use?
+this command might be able to help you for some recon for your secret locally. `kubectl get secret azureoperatorsettings -n operators -o go-template='{{range $k,$v := .data}}{{printf \"%s: \" $k}}{{if not $v}}{{$v}}{{else}}{{$v | base64decode}}{{end}}{{\"\n\"}}{{end}}'` [SO thread for reference](https://stackoverflow.com/questions/56909180/decoding-kubernetes-secret/58117444#58117444)
+
+## What if I see operator-namespace already exit for the cluster?
+
+If user will see already installed operator-namespace then it means user have tried installing the operator-namespace in-past, and it also means that user can now successfully move to the next step. 
+
+But, let me here add a small note that in the edge case of anyone playing around with out-of-order running of command, and wondering why the existing operator running is not 
+
+Minor detail of when steps are done out of sync and user see: 
+
+* `namespace already exist` - This means that azure-service-operator (ASO) is already been run for the cluster before.
+
+* `service-principal` issues, silent failures with service principal in use is expired or re-set in those case you can follow the as mentioned here https://docs.microsoft.com/en-us/cli/azure/create-an-azure-service-principal-azure-cli#create-a-service-principalor in step-4 here in official service operator page. https://operatorhub.io/operator/azure-service-operator
+
+* `kubectl api-resource` failure with `error: unable to retirve the com0plete list of server APIs: metcs.k8s.io/vabeta=1: the server is currently unable to handle the request.` - This error is a big indicaiton of network error within cluster and this is jsut the causal occurrence and not something ASO needs handles.
+
+ ## I see connection refuse for `cert-manager-package`?
+ One of the possible reason could be related to following open error: https://github.com/jetstack/cert-manager/issues/2752
```

Comment:
```suggestion
 ## When trying to install cert-manager, I'm seeing that the connection is refused for `cert-manager-package`?
This could be related to following open error: https://github.com/jetstack/cert-manager/issues/2752
```